precompiles/staking/types.go

```diff
@@ -149,6 +155,69 @@ func NewMsgCreateValidator(args []interface{}, denom string) (*stakingtypes.MsgC
 	return msg, delegatorAddress, nil
 }
 
+// NewMsgEditValidator creates a new MsgEditValidator instance and does sanity checks
+// on the given arguments before populating the message.
+func NewMsgEditValidator(args []interface{}) (*stakingtypes.MsgEditValidator, common.Address, error) {
+	if len(args) != 4 {
+		return nil, common.Address{}, fmt.Errorf(cmn.ErrInvalidNumberOfArgs, 4, len(args))
+	}
+
+	description := stakingtypes.Description{}
+	if descriptionInput, ok := args[0].(Description); ok {
+		description.Moniker = descriptionInput.Moniker
+		description.Identity = descriptionInput.Identity
+		description.Website = descriptionInput.Website
+		description.SecurityContact = descriptionInput.SecurityContact
+		description.Details = descriptionInput.Details
+	} else {
```

Comment:
it's better to use the opposite value for readability

```
descriptionInput, ok := args[0].(Description)
if !ok {
    return nil, common.Address{}, fmt.Errorf(cmn.ErrInvalidDescription, args[0])
}

// ...


```

Reply:
I think my method is not a big problem. When the decoding is correct, I will store the information in `description := stakingtypes.Description{}`, so that it can be directly filled in `msg := &stakingtypes.MsgEditValidator` later.

Reply:
Agree with @fedekunze here, also I think it might be cleaner using the `method.Inputs.Copy`

Reply:
ok, will refactor.